### PR TITLE
Fix public-url update (#119)

### PR DESCRIPTION
**Pull request by arturo-seijas**
Fixes the connection error preventing the public-url change to complete successfully

### CODE REVIEW

File: lib/charms/layer/jenkins/api.py [hunk 1]
```diff
@@ -52,11 +52,7 @@ def __init__(self, packages=None):
 
     @property
     def url(self):
-        config = hookenv.config()
-        prefix = urlparse(config["public-url"]).path
-        if len(prefix) > 0 and prefix[-1] != '/':
-            prefix += '/'
-        return urljoin("http://localhost:8080/", prefix)
+        return "http://localhost:8080/"
```
Comment (mthaddon) [inline lib/charms/layer/jenkins/api.py:55]: I'm not sure this is what we want. Currently jenkins can be configured to listen on a sub-path of a domain. Won't this mean that no longer works (per your test updates that sound like it's the case as well)?

Comment (arturo-seijas) [reply lib/charms/layer/jenkins/api.py:55]: When that prefix is set, Jenkins will still listen in localhost:8080. This means that prior to this change, when setting an URL with a prefix, the configuration change breaks as the charm is unable to reach the API resulting in a 404 error

Comment (mthaddon) [reply lib/charms/layer/jenkins/api.py:55]: As discussed IRL this seems to work when you deploy with config set, but not when you try to change it afterwards. There'll need to be some code updates to this PR to reflect that.

File: lib/charms/layer/jenkins/configuration.py [hunk 2]
```diff
@@ -50,56 +50,24 @@ def migrate(self):
...
-            if not found:
-                update = True
-
-        if update:
-            with open(paths.DEFAULTS_CONFIG_FILE, 'w') as defaults_file:
-                defaults_file.write(defaults_content + "\n")
-            return True
-
-        return False
+        # Since version 2.332.1 Jenkins is not loading env vars from the default config file
```
Comment (mthaddon) [inline lib/charms/layer/jenkins/configuration.py:65]: We'll need to check if they're using this version or later before taking this approach (unless we're sure this approach will work for older versions as well.

**Comment by mthaddon** [comment]
LGTM, thx.

**Pull request merged**